### ISSUE TITLE
`Domain` class does not have attribute `self._intent`

### ISSUE THREAD

**Issue by Ghostvv** [RasaHQ/rasa_core#831]
**Rasa Core version**: `master`

**Python version**: `3.6`

**Operating system** (windows, osx, ...): `osx`

**Issue**:
`Domain` class uses attribute `self._intent`, which is an attribute of its subclass `TemplateDomain`:
https://github.com/RasaHQ/rasa_core/blob/b956f7fc98ac463631381721099d49ccd9d9e57f/rasa_core/domain.py#L236

**Content of domain file** (if used & relevant):
```yaml

```

**Comment by Ghostvv** [comment]
@tmbo's suggestion: `Domain` should get a method like `def intents_with_properties()` and `TemplateDomain` should implement it. We can think about getting rid of `TemplateDomain`, as there isn't a reason anymore to split it, but `Domain` shouldn't access private properties of a subclass.

**Comment by jahid-ict** [comment]
I am also getting this and hence can't use new release. I have only a single intent for my bot. Is that the reason behind this?

**Comment by jahid-ict** [comment]
@akelad  . In this new 0.10.x release, my bots trained successfully. When I type a message, it extract my message nicely. But at the end, this shows the following KeyError. My Domain file has only one intent. For your kind information, if I run the the bot without using nlu file, it works good. Adding nlu file make this error. Any help is appreciated.
![image](https://user-images.githubusercontent.com/9244767/43778723-44d7c526-9a78-11e8-8484-5dfcb995a726.png)

**Comment by tmbo** [comment]
Is there some output before that, showing which intent it recognized?

**Comment by jahid-ict** [comment]
@tmbo  Bot shows the following in debug mode before showing the KeyError. FYI, Since, I have only one intent, then when I trained my bot, it seems that bot doesn't train intent classifier. I assume, that's the reason behind "none" intent.
![image](https://user-images.githubusercontent.com/9244767/43780533-9f37546a-9a7c-11e8-829c-87147bc03e83.png)

Output before showing KeyError.

![image](https://user-images.githubusercontent.com/9244767/43780238-f648422e-9a7b-11e8-9715-48227e625fd3.png)

**Comment by tmbo** [comment]
Yes, so the exception you are getting is because of an earlier issue (I've fixed the exception, but you will still get the None intent). The `None` intent is most likely due to you only having one intent class (the sklearn classifier for example won't train if there is only one class).

**Comment by jahid-ict** [comment]
@tmbo  `None` intent is not any problem as long as my bot works. But due to KeyError, my bot doesn't works. **What should I do now to make my bot working?** Any suggestion please? I am now working with rasa_core=0.9.8.

**Comment by tmbo** [comment]
I've created a PR to fix the issue, we'll review it and release a bug fix release to fix the issue in 0.9.9

**Comment by jahid-ict** [comment]
@tmbo . In 0.9.8, my bot works. Actually I am facing problem on rasa_core=0.10.x.

**Comment by tmbo** [comment]
Sorry my bad I meant 0.10.4
